status-im/status-react
# [#20804] linear gradient exception

fixes #20804

## Summary

This PR adds a safety wrapper for the Linear Gradient component, as we've had multiple issues due to invalid colors being passed to the component.

Now when an invalid color is found it's substituted by `"transparent"` so that the app doesn't crash, and an error with the given wrong colors is logged.

Additionally, this PR solves another problem: we have mistakenly named the `flamingo` color in our codebase, it's called as `pink` in figma (https://www.figma.com/design/v98g9ZiaSHYUdKWrbFg9eM/Foundations?node-id=619-10064&t=aYNJmP11jE8eqoh8-0)

![image](https://github.com/user-attachments/assets/5a849566-d8f6-46e7-9ad8-fce02d9160f3)

When we received `:pink` from status go it was resolved as `nil` so some components didn't render as expected (such as the `LinearGradient` crashing).

## Review notes

**IMPORTANT**
The `yin-yang` color is still not supported on mobile, but our accounts cards aren't crashing now :)

## Testing notes
- Please give a quick tour through the app, mostly focusing on places where the customization color is used.
- Before this PR, creating a profile on Desktop with the pink color and syncing it to mobile should be completely broken in terms of UI colors.

### Steps to test

- Create a Status profile
- Sync the profile on both Mobile and Desktop
- Create a watched account
- Change the color of the watched account to **pink** on Desktop
- On mobile, the wallet home account cards should look as expected
- Change the color of the watched account to **yin-yang* on Desktop
 - On mobile the account cards show transparent, but the app doesn't crash as before.

status: ready
